addon/components/detail-learning-materials.js

```diff
@@ -40,20 +40,13 @@ export default Component.extend(SortableByPosition, {
     return (!isManaging && !displayAddNewForm && !isSorting && editable);
   }),
 
-  proxyMaterials: computed('subject.learningMaterials.@each.position', function(){
-    return new RSVPPromise(resolve => {
-      let materialProxy = ObjectProxy.extend({
-        confirmRemoval: false,
-      });
-      this.get('subject').get('learningMaterials').then(materials => {
-        let sortedMaterials = materials.toArray().sort(this.get('positionSortingCallback'));
-        resolve(sortedMaterials.map(material => {
-          return materialProxy.create({
-            content: material
-          });
-        }));
-      });
-    });
+  proxyMaterials: computed('subject.learningMaterials.@each.position', async function() {
+    const materialProxy = ObjectProxy.extend({ confirmRemoval: false });
+    const materials = await this.subject.learningMaterials;
```

Comment:
This is an ember data model and should be `this.subject.get('learningMaterials');`